conjur/init_controller.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+
+"""
+InitController module
+
+This module is the controller that facilitates all init actions
+required to successfully configure the conjurrc
+"""
+
+# Builtin
+import logging
+import os
+import sys
+
+from urllib.parse import urlparse
+
+import OpenSSL
+
+class InitController:
+    """
+    InitController
+
+    This class represents the Presentation Layer for the init command
+    """
+    DEFAULT_CONFIG_FILE = os.path.expanduser(os.path.join('~', '.conjurrc'))
+
+    force_overwrite = False
+    init_input_obj = None
+    init_logic = None
+
+    def __init__(self, init_input_obj, init_logic, force):
+        self.init_input_obj = init_input_obj
+        self.init_logic = init_logic
+        self.force_overwrite = force
+
+    def call(self):
+        """
+        Method that facilitates all method calls in this class
+        """
+        fetched_certificate = self.get_server_certificate(self.init_logic,
+                                                          self.init_input_obj)
+        self.get_host_info(self.init_input_obj)
+        self.write_certificate(self.init_logic,
+                               self.init_input_obj,
+                               self.init_input_obj.cert_file,
+                               fetched_certificate)
+        self.init_logic.write_conjurrc(self.DEFAULT_CONFIG_FILE,
+                                       self.init_input_obj,
+                                       self.force_overwrite)
+
+    # pylint: disable=line-too-long
+    @staticmethod
+    def get_server_certificate(init_logic, input_obj):
+        """
+        Method to get the certificate from the Conjur endpoint detailed by the user
+        """
+        url = input_obj.appliance_url if input_obj.appliance_url else input("Enter the URL of your Conjur service: ")
+        input_obj.appliance_url = ''.join(url)
```

Comment:
I don't know what is being done here and why. Can you explain or add a comment?

Reply:
the `''.join()` casts the input to a string. I don't know `str()` didn't work but `''.join()` did the job

Reply:
Why would it be in a list?

Reply:
Because that is how argparse passes it to us. For example:
```
$ ./pkg_bin/conjur init --url localhost --account someaccount

Namespace(resource='init', debug=False, ssl_verify=True, account=['someaccount'], cert=None, force=False, url=['localhost'])
```

Reply:
This seems like something to look into - arg parsing should not give you this kind of input. Even if it does create this type of data, it should be turned into a string (or a URL object) before it reaches this class.

Reply:
Some of the input gets directly fed into the conjurrc object. For example, when `conjur init --url=https://localhost --account=someacc`
Otherwise, we interact with the user requesting this input. In this case, we then fill in the object and do the `''.join()`

Reply:
I think something is wrong here though as you should not be funneling raw CLI parsing output as the configuration object for the further steps down the line like the `Client`. Generally this is worked out via invocation of `myfunction(**vars(args))`. See here for more details as the code was using this previously iirc: https://www.peterbe.com/plog/vars-argparse-namespace-into-a-function.

Reply:
I added the suggestion to `Cli.run_client_action(resource, vars(args)` but I still get 

`{'resource': 'init', 'debug': False, 'ssl_verify': True, 'account': None, 'cert': None, 'force': False, 'url': ['https://localhost']}` if a user passes in url